pkg/testcase/pod.go

```diff
@@ -65,6 +75,28 @@ func getPrivatePods(cluster *shared.Cluster) (podDetails string) {
 	return podDetails
 }
 
+func CheckPodStatus(cluster *shared.Cluster) {
+	cmd := `kubectl get pods -A ` +
+		`-o jsonpath='{range .items[?(@.status.containerStatuses[-1:].state.waiting)]}{.metadata.name}: ` +
+		`{@.status.containerStatuses[*].state.waiting.reason}{"\n"}{end}'`
+	Eventually(func(g Gomega) bool {
+		pods, err := shared.GetPods(false)
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(pods).NotTo(BeEmpty())
+
+		res, _ := shared.RunCommandHost(cmd + " --kubeconfig=" + shared.KubeConfigFile)
+		if res != "" {
+			shared.LogLevel("info", "Waiting for pods: \n%s", res)
+			return false
+		}
+		for i := range pods {
+			processPodStatus(cluster, g, &pods[i], podAssertRestarts, podAssertReady)
+		}
+
+		return true
+	}, "600s", "10s").Should(BeTrue(), "failed to process pods status")
```

Comment:
Could we consider rephrasing it to 'Pod is not in the desired state,' since it seems that its the pod's status rather than the processing of it?

Reply:
Updated